gitpod-io/website
# Update Single Cluster Ref Arch AWS Workspace Node Storage to max throughput and IOPS

### Changes
1. Change to the max IOPS and throughput: This PR changes the recommended IOPS and throughput configuration for the storage (`gp3`) used for workspace nodes in the AWS version of the single cluster reference architecture to be the maximum possible (see [AWS docs](https://aws.amazon.com/ebs/general-purpose/)) for each. 
2. Increase to 360 GB storage size

## Reasoning
- We are currently expecting users to be able top run 9 Workspaces per node - see (internal) [assumptions](https://www.notion.so/gitpod/b34589d1df2349a6b6a7fe50a06e333f#9c7c3d7233b4429eb4c933722a3728b6)
- Given what we've seen in SaaS, and assuming a similar usage profile to the average SaaS user, the current IOPS and throughput are too low and we are increasing them to be as equivalent as possible to what we are running in SaaS (see related [internal discussion](https://gitpod.slack.com/archives/C02F19UUW6S/p1660144921528479?thread_ts=1660126413.044819&cid=C02F19UUW6S)). 
- This is only applied to workspace nodes because it is assumed that service nodes do not require this amount of throughput or IOPS. 
- Storage size is increased to 360 GB because: With 9 expected workspaces per node, and assuming 40 GB per workspace (30 gb for files, 10 gb for images) we get 360 GB. 


## Important
- This would increase the [currently projected cost](https://calculator.aws/#/estimate?id=d591c32d9457773170ca2e0f68d728fad0e0df19) by 299$ per month (see new [estimate](https://calculator.aws/#/estimate?id=4605b529918bbcde90694ee445e1a5dbd68a6bd9)) (the link to the pricing calculator in the docs page has been updated). 
- If approved, we will need to update the corresponding terraform module @nandajavarma 

## Related Issue(s)

Fixes #

## How to test
- Think through this change and evaluate whether it makes sense
- Ensure how this is done is sufficient - or do we need to change this anywhere else?

## Release Notes

```release-note
NONE
```

<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/2624"><img src="https://gitpod-staging.com/button/open-in-gitpod.svg"/></a>



<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2624"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
